### PR TITLE
feat: new rpc RawCheckpoints

### PR DESCRIPTION
**Pull request by gusin13**
### Description

This PR is part of optimizations done to improve babylon-api performance. 

The explorer calls `RawCheckpoint` rpc multiple times for each epoch.  I have added a new rpc `RawCheckpoints` which can ingest a range of epochs using pagination params. 

### Testing

Added fuzz, e2e tests and cli queries.

### CODE REVIEW

File: x/checkpointing/keeper/grpc_query_checkpoint.go [hunk 1]
```diff
@@ -60,6 +61,31 @@ func (k Keeper) RawCheckpoint(ctx context.Context, req *types.QueryRawCheckpoint
...
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "invalid request")
+	}
+
+	sdkCtx := sdk.UnwrapSDKContext(ctx)
+	store := k.CheckpointsState(sdkCtx).checkpoints
+
+	var checkpointList []*types.RawCheckpointWithMeta
+	pageRes, err := query.Paginate(store, req.Pagination, func(key, value []byte) error {
+		ckptWithMeta, err := types.BytesToCkptWithMeta(k.cdc, value)
```
Comment (vitsalis) [inline x/checkpointing/keeper/grpc_query_checkpoint.go:75]: Why not use unmarshal?

File: x/checkpointing/keeper/grpc_query_checkpoint_test.go [hunk 2]
```diff
@@ -47,6 +51,39 @@ func FuzzQueryEpoch(f *testing.F) {
 	})
 }
 
+func FuzzQueryRawCheckpoints(f *testing.F) {
+	datagen.AddRandomSeedsToFuzzer(f, 100)
```
Comment (vitsalis) [inline x/checkpointing/keeper/grpc_query_checkpoint_test.go:55]:
```suggestion
	datagen.AddRandomSeedsToFuzzer(f, 10)
```

We now do `10`

**Review by vitsalis — APPROVED** [review]

File: x/checkpointing/keeper/grpc_query_checkpoint.go (hunk 1 above)
Comment (gusin13) [reply x/checkpointing/keeper/grpc_query_checkpoint.go:75]: `BytesToCkptWithMeta`  unmarshals, its used in other places in code as well so used the same.

File: proto/babylon/checkpointing/v1/query.proto [hunk 3]
```diff
@@ -24,6 +24,13 @@ service Query {
         "/babylon/checkpointing/v1/raw_checkpoint/{epoch_num}";
   }
 
+  // RawCheckpoints queries a checkpoints at a given epoch number.
```
[the remark this answers is not in the document]
Comment (gusin13) [reply proto/babylon/checkpointing/v1/query.proto]: sure, was a typo, fixed now

File: x/checkpointing/keeper/grpc_query_checkpoint.go (hunk 1 above)
Comment (gusin13) [reply x/checkpointing/keeper/grpc_query_checkpoint.go]: changed to `MustUnmarshal` as per our offline discussion.

**Pull request merged**